### ISSUE TITLE
Properties of Carbon Document being deleted on independent queries

### ISSUE THREAD

**Issue by Yahaire** [CarbonLDP/carbonldp-js-sdk#351]
I am building a page where you can see the details of a movie and a summary of related movies. The documents are got as so:
```typescript
const movie = await this.moviesService.getMovieById(movieId);
const similarMovies = await this.moviesService.getRelatedMoviesByMovie(movieId, 4);
```
It seems like this is causing the `movie` const to lose some of its properties. From the test I've done, it looks like, even though, the objects `movie` and `similarMovies` are assigned in different queries, the document related to `movie` is found on the relatedMoviesResults (Which discards it before returning it to `similarMovies`). This results in the full `movie` document obtained from the `getMovieById` method (A simple wrapper for Carbon's `.$get` method) being replaced by a partial document found on the `getRelatedMovies` method (Which implements Carbon's `$getMembers`.

As a quick workaround, I am using the following lines which cause the `movies` object to be replaced once again by the document with full data:
```typescript
let movie = await this.moviesService.getMovieById(movieId);
const similarMovies = await this.moviesService.getRelatedMoviesByMovie(movieId, 4);
movie = await this.moviesService.getMovieById(movieId);
```

Guessing that Carbon has a shared cache across all of its queries which is what's causing this. Is there a way to indicate Carbon to separate cache by queries or a workaround you would suggest for my case in order to not have to do two requests each time I want to make sure I have the full movie details?

**Comment by roddolf** [comment]
Yes, de SDK have a cache that shares the reference of all the resources requested.
And if you query a resource from different places, the data will be combined.

If this is not working, can you provided more information about your queries so we can replicate the bug?

**Comment by roddolf** [comment]
Thank you for the extended code of your queries.

Unfortunately, I'm not able to reproduce the bug.
 I get the behavior of combining the data with every query executed.
